quickwit/quickwit-common/src/type_map.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2023 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::any::{Any, TypeId};
+use std::collections::HashMap;
+
+#[derive(Debug, Default)]
+pub struct TypeMap(HashMap<TypeId, Box<dyn Any + Send + Sync + 'static>>);
+
+impl TypeMap {
+    pub fn contains<T: Any + Send + Sync + 'static>(&self) -> bool {
```

Comment:
I think you can remove `static` as it's already insured by `Any`.

Reply:
Cool.